### ISSUE TITLE
CSS charter 2019

### ISSUE THREAD

**Issue by xfq** [w3c/strategy#188]
https://w3c.github.io/charter-drafts/css-2019.html

**Comment by xfq** [comment]
I'm currently working on the charter. Will circulate it to the WG and request horizontal review.

**Comment by r12a** [comment]
The long alphabetical list of modules is suboptimal for me. It would be better to group, eg. 
- WD
  - with target date  for next state
  - without
- CR
  - with target date for next state
  - without

Also, what do 'Adopted' and 'Reference' mean here?  Please clarify in the charter.

**Comment by r12a** [comment]
> More detailed milestones and updated publication schedules are available on the group publication status page.

The link appears to be broken.

**Comment by xfq** [comment]
> Also, what do 'Adopted' and 'Reference' mean here? Please clarify in the charter.

It's defined in https://github.com/w3c/charter-drafts/blob/gh-pages/draft-states.md and https://www.w3.org/2019/Process-20190301/#WGCharter

Would it be helpful by linking the text to the corresponding definitions in the Process?

**Comment by xfq** [comment]
> > More detailed milestones and updated publication schedules are available on the group publication status page.
> 
> The link appears to be broken.

Fixed. Thanks.

**Comment by michael-n-cooper** [comment]
I'm glad they used the suggested boilerplate phrase "Each specification should contain a section on accessibility that describes the benefits and impacts, including ways specification features can be used to address them, and recommendations for maximising accessibility in implementations."

However, the draft charter does not otherwise reference accessibility liaison, outside the stock horizontal review statement. The previous charter had "In addition to the above catch-all reference to horizontal review which includes accessibility review, this Working Group will work with the Accessible Rich Internet Applications Working Group to produce CSS Accessibility API Mappings draft as well as best practices of authoring accessible documents with CSS. A taskforce will be the primary venue for this work, and those specifications will be published jointly."

I would like to see either that paragraph restored, or a liaison statement to APA added with similar text. The text I suggest for a liaison statement (this could be adapted into the above paragraph restoration if that approach is preferred) is: "Coordinate on features of CSS that impact accessibility to people with disabilities. This involves in particular work on accessible navigation which needs to be addressed coherently across multiple specifications. It is also involves localized accessibility issues related to the features of individual specifications, and the development of new CSS specifications to address accessibility use cases where appropriate. Some of this work may be carried out in the joint CSS Accessibility Task Force.".

**Comment by xfq** [comment]
Thanks @michael-n-cooper. I have restored the paragraph here: https://w3c.github.io/charter-drafts/css-2019.html#coordination

**Comment by michael-n-cooper** [comment]
Thanks. I think updating the paragraph would be useful, as the focus has shifted, though I'm happy to see it in any form. :) In particular I think it should reference APA rather than ARIA. Specific wording I suggest is:

In addition to the above catch-all reference to horizontal review which includes accessibility review, this Working Group will work with the [Accessible Platform Architectures Working Group](https://www.w3.org/WAI/APA/) to work on accessible navigation which needs to be addressed coherently across multiple specifications, address accessibility issues related to the features of individual specifications, and develop new CSS specifications to address accessibility use cases where appropriate. Some of this work may be carried out in the joint [CSS Accessibility Task Force](https://www.w3.org/WAI/APA/task-forces/css-a11y/).

**Comment by xfq** [comment]
Looks good to me. I just updated the paragraph.

**Comment by michael-n-cooper** [comment]
Thanks. My review is complete, over to @brewerj to complete accessibility review sign-off.

**Comment by svgeesus** [comment]
@r12a wrote:

> Also, what do 'Adopted' and 'Reference' mean here? Please clarify in the charter.

Those terms should be in *all* charters, as required by Process 2019. If you haven't seen them before, it either means you haven't reviewed a charter since the new Process came into effect or, perhaps, that any charter you did review did not use the current charter template

**Comment by r12a** [comment]
I would like to see a target date for Writing Modes 4. We need there to be some pressure on implementers to implement the `sideways` properties because international users need them. These were removed from WM 3 to allow the rest of the WM module to move quickly to Rec, and there was an expectation set that only a little extra time was needed for implementation of the WM4 features. So i'm a bit surprised that we are to wait still another 6 months from now for WM3 (which international users also need, and are beginning, might i observe, to despair about), and that there's no planned date for WM4 at all.

**Comment by r12a** [comment]
CSS Counter Styles Level 3 also has no target completion date, and has been becalmed for some time now, even though there was some mention a while back that Chrome was expecting to provide a second implementation.  This is also something that will benefit international communities, but that has acquired an air of neglect.

**Comment by astearns** [comment]
Completion dates are not being held back on implementations or even interoperability. Both of those things are happening more quickly than moving specs to REC. The main block is completing a test suite and providing a test report. There is very little investment in finishing test suites, so most CSS modules languish in a mostly-done, implemented-well-enough-to-use, not-tested-well-enough-to-finish state. I have not yet seen that setting deadlines for specs changes the testing problem.

**Comment by r12a** [comment]
i18n also hopes to see the following working drafts proceed at least to CR during the upcoming charter period, and preferably to Rec, in order to improve international use of the Web.  None of them have target dates in the charter, so the implication seems to be that no significant progress will be made for at least 3 more years, after which we still face the prospect of a long CR period.  So we appear to be looking at 5 or more years before completion.  

Specs with a key i18n interest, and for which international users have been waiting, and currently need, but which have no target date, include: CSS Text Decoration Module Level 4, CSS Inline Layout Module Level 3, CSS Logical Properties and Values Level 1, CSS Ruby Layout Module Level 1, CSS Text Module Level 3 (14 years), CSS Text Module Level 4.

From a worldwide Web perspective, and given that we have already been waiting some years for progress on these specs (for many of the features specified it's over 10 years already), i'm sorry but i don't really feel i would be honest if i said that I'm happy with this charter.

**Comment by r12a** [comment]
> There are currently 28 open and 27 closed issues and the test suite currenly has 154 of 193 required tests meet CR exit criteria - Blink passes 74%, Gecko 99%.

The link you used is for the CSS Lists module. The Counter Styles module has [9 open issues](https://github.com/w3c/csswg-drafts/labels/css-counter-styles-3), though i think at least 2-3 of those can be closed right away. What i'm asking for is that time be allocated to close out the remainder.

> Help is always welcome to resolve issues, file bugs on implementations, and create new tests.

I don't know which of the implications of that statement i should be taking away as the message. For the record,  I'd argue that the i18n WG already contributed significantly to the development of the spec, including discussions that are not reflected in the list of GH issues.  I personally provided around 80% of your 154 tests (not to mention the larger set for the [Ready-made Counter Styles](https://www.w3.org/TR/predefined-counter-styles/) document we created to support this spec).  The i18n group have also been doing the work of raising and promoting [bugs on implementations](https://github.com/w3c/i18n-activity/issues/170) for all major browser engines.  (I even created [an app](https://r12a.github.io/app-counters/) that implements the counter-styles in the spec so that people can test stuff.) In short, i think we have provided a significant amount of help already, but this is not a deliverable of the i18n WG.

> Making up a reassuring date in the future, based on mere prognostication, in the certain knowledge that the date cannot be predicted, helps no-one.

I'm not asking anyone to make up  "a reassuring date in the future, based on mere prognostication, in the certain knowledge that the date cannot be predicted".  I'm asking that the CSS WG figure out what needs to be done to bring this spec to Rec, and then set itself a target to work towards in order to get it done, enlisting help where needed.  

This spec was first published as a WD almost 7 years ago (though it's history goes back much further). It has been in CR for over 4 years. It's not a hugely complicated spec, and the work seems pretty much done. There is a need for its features.  But there appears to be no plan in place to make those features available to users in an interoperable fashion within the next 3 years.  IMO that's what helps no-one.
